src/components/Counter/index.js

```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react'
+import './CounterStyles.css'
+import Button from '../Button'
+
+class Counter extends Component {
+
+  state = {
+    count: 0
+  }
+
+  render() {
+    const increment = () => {
+      this.setState({ count: this.state.count + 1 })
```

Comment:
if you'd like to access the previous state in the `setState` method to keep your state consistent and async-safe - it's better to pass a callback and use prevState reference:
```suggestion
      this.setState(prevState => ({ count: prevState.count + 1 }))
```

Reply:
didn't know awesome thanks